### PR TITLE
Show events of a configurable amount of past days

### PR DESCRIPTION
**Pull request by Tom-Hirschberger**
Hi,

want to include a birthday calendar to my mirror which shows upcoming birthdays and as a reminder birthdays of the last two days.
I used [MMM-CalendarExt2](https://github.com/MMM-CalendarExt2/MMM-CalendarExt2) for this job in the past but the module is not supported any more and very complicated to configure.
I managed to style the default calendar module to my needs but what i am missing is to display already past events within a configurable time range.

I included the translations of "YESTERDAY" and "DAYBEFOREYESTERDAY" to all translation files and modified the code to accept a new option `pastDaysCount` which controls of how many days past events should be displayed.

### CODE REVIEW

File: modules/default/calendar/calendar.js [hunk 1]
```diff
@@ -232,7 +234,10 @@ Module.register("calendar", {
 					const dateRow = document.createElement("tr");
 					dateRow.className = "dateheader normal";
 					if (event.today) dateRow.className += " today";
+					else if (event.dayBeforeYesterday) dateRow.className += " dayBeforeYesterday";
+					else if (event.yesterday) dateRow.className += " yesterday";
 					else if (event.tomorrow) dateRow.className += " tomorrow";
+					else if (event.dayAfterTomorrow) dateRow.className += " dayAfterTomorrow";
```
Comment (rejas) [inline modules/default/calendar/calendar.js:240]: Can you add tests for these classnames in tests/electron/modules/calendar_spec.js ?

File: translations/en.json [hunk 2]
```diff
@@ -1,6 +1,7 @@
 {
 	"LOADING": "Loading …",
 
+	"YESTERDAY": "Yesterday",
```
Comment (rejas) [inline translations/en.json:4]: So, what are english displays showing then?

**Review by rejas — CHANGES_REQUESTED** [review]
Nice PR, just would be nice to have some tests

File: translations/en.json (hunk 2 above)
Comment (Tom-Hirschberger) [reply translations/en.json:4]: They will show "Yesterday" like "Today" for today

File: modules/default/calendar/calendar.js (hunk 1 above)
Comment (Tom-Hirschberger) [reply modules/default/calendar/calendar.js:240]: Hi, i will try to. I struggled with the last time i opened a pull request and failed but this time new chance.

**Comment by sdetweil** [comment]
I'm not fond of this pr.  we should leave this to extension modules and not try to jam past events into the default display.

have enough calendar problems as it is

File: translations/af.json [hunk 3]
```diff
@@ -1,6 +1,7 @@
 {
 	"LOADING": "Besig om te laai …",
```
Comment (KristjanESPERANTO) [inline translations/af.json:3]: Is it intentional that some languages are missing DAYBEFOREYESTERDAY?

Comment (Tom-Hirschberger) [reply translations/af.json:3]: There is no useful translation for "DAYBEFOREYESTERDAY" and "DAYAFTERTOMORROW" in some languages. The calendar module takes care of these exceptions for "DAYAFTERTOMORROW" already.
I adapted the mechanism to "DAYBEFOREYESTERDAY"

Comment (KristjanESPERANTO) [reply translations/af.json:3]: Ah okay, thanks for the explanation!

I think this PR is a useful addition to the standard module 🙂

**Comment by rejas** [comment]
> 

I agree with you that the calendar module is bloated / full-of-old-code etc 

But this PR seems small and straightforward enough IMO to be merged. No weird edge cases or so... I am in favor of merging. Whats your take @khassel ?

**Comment by khassel** [comment]
It's a valid usecase so I will not block a merge. But on the other side it does not simplify the code. Because it is implemented as a new option it hopefully will not increase support.

**Comment by sdetweil** [comment]
it will increase support.  layout, colors,. blah blah..

**Comment by Tom-Hirschberger** [comment]
Hi, i can understand the concerns about increasing amount of support with increasing amount of features.
What i do not understand is why the introduction of past events (with only a small amount of changes needed) will increase the support amount for colors and the basic layout.

File: translations/en.json (hunk 2 above)
Comment (Tom-Hirschberger) [reply translations/en.json:4]: I added some checks to prove that the CSS classes got added correctly which was more or less copy and paste. I will try to add more changes but i will need to dive deeper into into the test classes to check the locals

File: modules/default/calendar/calendar.js (hunk 1 above)
Comment (Tom-Hirschberger) [reply modules/default/calendar/calendar.js:240]: I added some checks to prove that the CSS classes got added correctly which was more or less copy and paste. I will try to add more changes but i will need to dive deeper into into the test classes to check the locals

**Comment by sdetweil** [comment]
based on past experience. more than one someone will want these past events formatted differently than current. and will want more, and might want them after the upcoming

**Comment by Tom-Hirschberger** [comment]
@sdetweil I wrote a transformer for MMM-CalenderExt3Agend today and did some hardcore CSS restyling to the module. As the module displays past events in default it is okay for me if do not want to merge the pull request.

But i will need to open a issue as the default calender module broadcasts the birthday as monthly events instead of yearly.

**Comment by sdetweil** [comment]
monthly vs yearly. but they are yearly..... why mangkey the data

**Review by rejas — APPROVED** [review]

**Comment by rejas** [comment]
Merging it since the majority seems to be in favor of it.

**Comment by rejas** [comment]
Could you please create a PR with the next options for the documentation page https://github.com/MichMich/MagicMirror-Documentation

@Tom-Hirschberger  ?

**Comment by Tom-Hirschberger** [comment]
Sure...
[https://github.com/MichMich/MagicMirror-Documentation/pull/156](https://github.com/MichMich/MagicMirror-Documentation/pull/156)

**Pull request merged**